packages/webapp/pages/account/security.tsx

```diff
@@ -0,0 +1,222 @@
+import { Button } from '@dailydotdev/shared/src/components/buttons/Button';
+import AccountDangerZone from '@dailydotdev/shared/src/components/profile/AccountDangerZone';
+import MailIcon from '@dailydotdev/shared/src/components/icons/Mail';
+import AuthContext from '@dailydotdev/shared/src/contexts/AuthContext';
+import LockIcon from '@dailydotdev/shared/src/components/icons/Lock';
+import { Overlay } from '@dailydotdev/shared/src/components/utilities';
+import React, { ReactElement, useContext, useState } from 'react';
+import {
+  AccountContentHeading,
+  AccountPageContainer,
+  AccountTextField,
+  ContentHeading,
+  ContentText,
+  OverlayContainer,
+  OverlayText,
+} from '../../components/layouts/AccountLayout/common';
+import EmailForm from '../../components/layouts/AccountLayout/EmailForm';
+import { getAccountLayout } from '../../components/layouts/AccountLayout';
+
+const socialProvider = getProviderMapClone();
+socialProvider.gitHub.style = { backgroundColor: '#383C47' };
+socialProvider.apple.style = { backgroundColor: '#404551' };
+const providers = Object.values(socialProvider);
+
+enum Display {
+  Default = 'default',
+  ChangeEmail = 'change_email',
+  ConnectEmail = 'connect_email',
+}
+
+const AccountSecurityPage = (): ReactElement => {
+  const { user } = useContext(AuthContext);
+  const [email, setEmail] = useState<string>();
+  const [emailSent, setEmailSent] = useState(false);
+  const [resetPasswordSent, setResetPasswordSent] = useState(false);
+  const [activeDisplay, setActiveDisplay] = useState(Display.Default);
+
+  if (!user) {
+    return null;
+  }
+
+  const onDelete = () => {};
+  const onChangeEmail = () => {
+    setEmailSent(true);
+    setActiveDisplay(Display.Default);
+  };
+
+  const onConnectEmail = () => {
+    setEmailSent(true);
+    setActiveDisplay(Display.Default);
+  };
+
+  const renderEmailAction = () => {
+    if (emailSent) {
+      return (
+        <OverlayContainer className="mt-6 border-theme-status-warning">
+          <Overlay className="bg-overlay-quaternary-bun" />
+          <p>
+            We sent an email to verify your account. Please check your spam
+            folder if you {`don't`} see the email.
+          </p>
+          <span className="flex flex-row gap-4 mt-4">
+            <Button buttonSize="xsmall" className="w-fit btn-primary">
+              Resend
+            </Button>
+            <Button buttonSize="xsmall" className="w-fit btn-secondary">
+              Cancel Request
+            </Button>
+          </span>
+        </OverlayContainer>
+      );
+    }
+
+    return (
+      // return user.password ? ( // temporary enabling
+      <Button
+        className="mt-6 w-fit btn-secondary"
+        onClick={() => setActiveDisplay(Display.ChangeEmail)}
+      >
+        Change email
+      </Button>
+    );
+  };
+
+  return (
+    <TabContainer showHeader={false} controlledActive={activeDisplay}>
```

Comment:
We can probably migrate these to separate individual components

Reply:
100% agree with that 🚢

Reply:
Added a task for it.